### PR TITLE
chore(ingest): add vulture config and remove some dead code

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py [hunk 1]
```diff
@@ -4,22 +4,11 @@
...
     SnowflakeConfig,
     SnowflakeProvisionRoleConfig,
 )
-from datahub.ingestion.source_config.usage.snowflake_usage import (
-    SnowflakeStatefulIngestionConfig,
-    SnowflakeUsageConfig,
-)
-
-
-class SnowflakeV2StatefulIngestionConfig(
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py:18]: @mayurinehate : can you confirm that this is safe to remove.

File: metadata-ingestion/src/datahub/ingestion/api/ingestion_state_provider.py [hunk 2]
```diff
@@ -1,31 +0,0 @@
-from abc import ABC, abstractmethod
-from typing import Any, Dict, NewType, Optional
-
-from datahub.ingestion.api.common import PipelineContext
-from datahub.metadata.schema_classes import DatahubIngestionCheckpointClass
-
-JobId = NewType("JobId", str)
-
-
-class IngestionStateProvider(ABC):
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/api/ingestion_state_provider.py:10]: @rslanka : can you confirm that this is safe to remove?

File: metadata-ingestion/src/datahub/configuration/source_common.py [hunk 3]
```diff
@@ -53,18 +53,6 @@ class DatasetSourceConfigBase(PlatformSourceConfigBase, EnvBasedSourceConfigBase
     """
 
 
-class EnvPlatformInstance(PlatformSourceConfigBase):
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/configuration/source_common.py:56]: 👍

File: metadata-ingestion/src/datahub/ingestion/source/looker_common.py [hunk 4]
```diff
@@ -66,12 +66,6 @@
 logger = logging.getLogger(__name__)
 
 
-def remove_view_from_field_name(field):
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/looker_common.py:69]: 👍

**Comment by shirshanka** [comment]
Waiting to approve until we get confirmations from the authors (@mayurinehate , @rslanka ) that these classes are safe to remove or whether they missed adding a dependency on them.

File: metadata-ingestion/src/datahub/ingestion/api/ingestion_state_provider.py (hunk 2 above)
Comment (rslanka) [reply metadata-ingestion/src/datahub/ingestion/api/ingestion_state_provider.py:10]: Yes, this is safe to remove. This got replaced by [ingestion_job_state_provider.py](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/api/ingestion_job_state_provider.py).

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py:18]: yes, it is safe to remove.

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**